### ISSUE TITLE
install the flash-attention

### ISSUE THREAD

**Issue by SISTMrL** [Dao-AILab/flash-attention#406]
hello, i run the pip install flash-attn, but it fails. the gpu i used is tesla m40, can the flash attn be installed on m40?

**Comment by tridao** [comment]
No, m40 is of [Maxwell](https://en.wikipedia.org/wiki/CUDA#GPUs_supported) generation (sm_52). FlashAttention right now supports Ampere (sm_80) and after.

**Comment by twmht** [comment]
@tridao 

Can you explain why it does not support maxwell? Are there any technical limitations?

**Comment by tridao** [comment]
> @tridao
> 
> Can you explain why it does not support maxwell? Are there any technical limitations?

Because it takes work. Maxwell does not have tensor cores. Someone would need to implement a version that doesn't use tensor cores.
